pkg/virt-handler/node-labeller/cpu_plugin.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return err
+	}
+
+	usableModels := make([]string, 0)
+	for _, mode := range hostDomCapabilities.CPU.Mode {
+		for _, model := range mode.Model {
+			if model.Usable == isUnusable || model.Usable == "" {
+				continue
+			}
+			usableModels = append(usableModels, model.Name)
+		}
+	}
+
+	n.hostCapabilities.items = usableModels
+
+	return nil
+}
+
+//loadHostSupportedFeatures loads supported features
+func (n *NodeLabeller) loadHostSupportedFeatures() error {
+	featuresFile := path.Join(nodeLabellerVolumePath + "supported_features.xml")
+
+	hostFeatures := SupportedHostFeature{}
+	err := n.getStructureFromXMLFile(featuresFile, &hostFeatures)
+	if err != nil {
+		return err
+	}
+
+	usableFeatures := make([]string, 0)
+	for _, f := range hostFeatures.Feature {
+		if f.Policy != RequirePolicy {
+			continue
+		}
+
+		usableFeatures = append(usableFeatures, f.Name)
+	}
+
+	n.supportedFeatures = usableFeatures
+	return nil
+}
+
+//loadCPUInfo load info about all cpu models
+func (n *NodeLabeller) loadCPUInfo() error {
+	files, err := ioutil.ReadDir(path.Join(nodeLabellerVolumePath, "cpu_map"))
+	if err != nil {
+		return err
+	}
+
+	models := make(map[string]cpuFeatures)
+	for _, f := range files {
+		fileName := f.Name()
+		if strings.HasPrefix(fileName, "x86_") {
+			features, err := n.loadFeatures(fileName)
+			if err != nil {
+				return err
+			}
+			cpuName := strings.TrimSuffix(strings.TrimPrefix(fileName, "x86_"), ".xml")
+
+			models[cpuName] = features
+		}
+	}
+
+	n.cpuInfo.models = models
+
+	return nil
+}
+
+func (n *NodeLabeller) getDomCapabilities() (HostDomCapabilities, error) {
+	domCapabilitiesFile := path.Join(nodeLabellerVolumePath + "virsh_domcapabilities.xml")
+	hostDomCapabilities := HostDomCapabilities{}
+	err := n.getStructureFromXMLFile(domCapabilitiesFile, &hostDomCapabilities)
+
+	return hostDomCapabilities, err
+}
+
+//LoadFeatures loads features for given cpu name
+func (n *NodeLabeller) loadFeatures(fileName string) (cpuFeatures, error) {
+	if fileName == "" {
+		return nil, fmt.Errorf("file name can't be empty")
+	}
+
+	cpuFeaturepath := getPathCPUFeatures(fileName)
+	features := FeatureModel{}
+	err := n.getStructureFromXMLFile(cpuFeaturepath, &features)
+	if err != nil {
+		return nil, err
+	}
+
+	modelFeatures := cpuFeatures{}
+	for _, f := range features.Model.Features {
+		modelFeatures[f.Name] = true
+	}
+	return modelFeatures, nil
+}
+
+//getPathCPUFeatures creates path where folder with cpu models is
+func getPathCPUFeatures(name string) string {
+	return path.Join(nodeLabellerVolumePath, "cpu_map", name)
+}
+
+//GetStructureFromXMLFile load data from xml file and unmarshals them into given structure
+//Given structure has to be pointer
+func (n *NodeLabeller) getStructureFromXMLFile(path string, structure interface{}) error {
+	rawFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	n.logger.V(4).Infof("loaded file: %#v", string(rawFile))
```

Comment:
we should be more explicit here. maybe just log what you wrote in the comment: "loading data from xml file"